File: abc/092/c.rb
n = gets.to_i
a = gets.split.map(&:to_i)

a.unshift(0).push(0)
tot_d = a.each_cons(2).reduce(0) { |sum, (x, y)| sum + (y - x).abs }

(1..n).each do |i|
  puts tot_d + (a[i+1] - a[i-1]).abs - (a[i+1] - a[i]).abs - (a[i] - a[i-1]).abs
end

File: abc/092/c.py
n = int(input())
a = list(map(int, input().split()))
a = [0] + a + [0]

sum = 0
for i in range(len(a)-1):
    sum += abs(a[i+1] - a[i])

for i in range(1, len(a)-1):
    if (a[i] - a[i-1]) * (a[i+1] - a[i]) > 0:
        print(sum)
    else:
        save = 2 * min(abs(a[i] - a[i-1]), abs(a[i+1] - a[i]))
        print(sum - save)
